### DISCUSSION TITLE
How should we deal with redundant branches?

### DISCUSSION THREAD

**Discussion by smol-ninja** [sablier-labs/flow#65] [General]
In open-ended, since there is no concept of "operators" or "approved third party", should we remove test branches that deal with malicious third party as the caller?

Take the following example from [`cancel.tree`](https://github.com/sablier-labs/v2-open-ended/blob/main/test/integration/cancel/cancel.tree):

```tree
cancel.t.sol
// --- snip --- //
          ├── when the caller is unauthorized
          │  ├── when the caller is the recipient
          │  │  └── it should revert
          │  └── when the caller is a malicious third party
          │     └── it should revert
          └── when the caller is authorized
// --- snip --- //
```

However, we could just say

```tree
cancel.t.sol
// --- snip --- //
          ├── when the caller is not sender
          │  └── it should revert
          └── when the caller is sender
// --- snip --- //
```

It simplifies the integration test and makes it clearer who the caller is.

What do you think @andreivladbrg ?

**Comment by smol-ninja** [comment]
@andreivladbrg would love to have your take on this as well.

**Comment by andreivladbrg** [comment]
when we talk about a "malicious third party" it doesn't necessarily refer to any operator or any approved address - it simply refers to an unknown actor within our protocol (i.e. not the `recipient` or the `sender`).

we will be adding NFTs, so we will have operators if you have a problem with the terminology.
